### PR TITLE
Issue with column sorting

### PR DESCRIPTION
**Pull request by andshk**

### CODE REVIEW

File: VirtoCommerce.CatalogModule.Data/Services/CatalogSearchServiceImpl.cs [hunk 1]
```diff
@@ -180,7 +180,30 @@ protected virtual void SearchCatalogs(SearchCriteria criteria, SearchResult resu
...
+                case "isVirtual:asc":
+                    return catalogs.OrderBy(x => x.IsVirtual).ToList();
+                case "isVirtual:desc":
+                    return catalogs.OrderByDescending(x => x.IsVirtual).ToList();
+                case "id:asc":
+                    return catalogs.OrderBy(x => x.Id).ToList();
+                case "id:desc":
+                    return catalogs.OrderByDescending(x => x.Id).ToList();
+                default:
+                    throw new InvalidProgramException("SearchCriteria.Sort has improper value");
```
Comment (artem-dudarev) [inline VirtoCommerce.CatalogModule.Data/Services/CatalogSearchServiceImpl.cs]: https://msdn.microsoft.com/ru-ru/library/system.invalidprogramexception(v=vs.95).aspx
The exception that is thrown when a program contains invalid Microsoft intermediate language (MSIL) or metadata. Generally this indicates a bug in the compiler that generated the program.

File: VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleCatalogsController.cs [hunk 2]
```diff
@@ -32,11 +32,12 @@ public CatalogModuleCatalogsController(ICatalogService catalogService, ICatalogS
 		[HttpGet]
         [Route("")]
         [ResponseType(typeof(webModel.Catalog[]))]
-        public IHttpActionResult GetCatalogs()
+        public IHttpActionResult GetCatalogs([FromUri]webModel.SearchOptions options)
```
Comment (tatarincev) [inline VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleCatalogsController.cs]: The formal parameter should be a type  SearchCriteria instead webModel.SearchOptions

File: VirtoCommerce.CatalogModule.Web/Model/SearchOptions.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿namespace VirtoCommerce.CatalogModule.Web.Model
```
Comment (tatarincev) [inline VirtoCommerce.CatalogModule.Web/Model/SearchOptions.cs]: Redundant type, need to remove it

File: VirtoCommerce.CatalogModule.Data/Services/CatalogSearchServiceImpl.cs [hunk 4]
```diff
@@ -180,7 +180,27 @@ protected virtual void SearchCatalogs(SearchCriteria criteria, SearchResult resu
                     }
                 });
 
-                result.Catalogs = catalogs.OrderBy(x => x.Name).ToList();
+                result.Catalogs = OrderItems(catalogs, criteria);
```
Comment (tatarincev) [inline VirtoCommerce.CatalogModule.Data/Services/CatalogSearchServiceImpl.cs]: use follow  code  for  sorting  instead switch case
` var sortInfos = criteria.SortInfos;
                if (sortInfos.IsNullOrEmpty())
                {
                    sortInfos = new[] { new SortInfo { SortColumn = "Name" } };
                }
                query = query.OrderBySortInfos(sortInfos);
`

**Pull request merged**